operate/schema/src/test/java/io/camunda/operate/util/OpensearchUtilTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.operate.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import io.camunda.operate.schema.templates.TemplateDescriptor;
+import io.camunda.operate.store.opensearch.client.sync.OpenSearchDocumentOperations;
+import io.camunda.operate.store.opensearch.client.sync.RichOpenSearchClient;
+import io.camunda.operate.store.opensearch.dsl.RequestDSL.QueryType;
+import jakarta.json.stream.JsonGenerator;
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.client.json.JsonpSerializable;
+import org.opensearch.client.json.jsonb.JsonbJsonpMapper;
+import org.opensearch.client.opensearch.core.SearchRequest;
+import org.opensearch.client.opensearch.core.SearchResponse;
+
+@ExtendWith(MockitoExtension.class)
+class OpensearchUtilTest {
+  private static final String INDEX_NAME = "list-view-index";
+  private static final String ALIAS_NAME = "list-view-alias";
+  private static final String DOC_ID = "123";
+
+  @Mock private TemplateDescriptor templateDescr;
+  @Mock private RichOpenSearchClient osClient;
+
+  @Mock private OpenSearchDocumentOperations docOperations;
+
+  @Mock private SearchResponse searchResponse;
+  @Captor private ArgumentCaptor<SearchRequest.Builder> searchRequestCaptor;
+
+  @BeforeEach
+  void setup() {
+    lenient().when(templateDescr.getFullQualifiedName()).thenReturn(INDEX_NAME);
+    lenient().when(templateDescr.getAlias()).thenReturn(ALIAS_NAME);
+    when(osClient.doc()).thenReturn(docOperations);
+  }
+
+  @Test
+  void shouldReturnSourceWhenCallingGetByIdAndGetResultExists() {
+    when(docOperations.getWithRetries(any(), any(), any(), any()))
+        .thenReturn(Optional.of(Map.of("treePath", "/a/b/c")));
+
+    assertThat(
+            OpensearchUtil.getByIdOrSearchArchives(
+                osClient, templateDescr, DOC_ID, QueryType.ONLY_RUNTIME, "treePath"))
+        .isEqualTo(Map.of("treePath", "/a/b/c"));
+
+    verifyGetRequest();
+    verifyNoMoreInteractions(docOperations);
```

Comment:
🔧 NABD but we generally split tests with `// given`/`// when`/`// then` comments to break it into the appropriate sections. Please don't consider this a change request, more just an FYI that you might also find useful/conventional